src/DelimitedDataParser/Exporter.cs

```diff
@@ -30,11 +30,17 @@ public class Exporter
         private bool _useExtendedPropertyForColumnName;
         private string _extendedPropertyKey;
 
+        private IProgress<int> _progress;
+
         /// <summary>
         /// Initializes a new instance of the <see cref="Exporter"/> class.
         /// </summary>
-        public Exporter()
+        /// <param name="progress">
+        /// The provider for receiving progress updates.
+        /// </param>
+        public Exporter(IProgress<int> progress = null)
```

Comment:
Method overloads with optional parameters [can cause runtime errors](https://haacked.com/archive/2010/08/10/versioning-issues-with-optional-arguments.aspx/). I'm not sure if the same applies to constructors, but I'd assume it does. We should either introduce a new constructor overload that takes an `IProgress`, rather than modify the existing default constructor, or prove that adding an optional parameter doesn't cause and issues.